src/shared/ui/survey/StackedCheckboxItem/StackedCheckboxItem.tsx

```diff
@@ -1,46 +1,133 @@
-import { FC } from 'react';
+import { FC, useMemo } from 'react';
 import { StyleSheet } from 'react-native';
 
-import StackedCheckboxConfig from './types';
-import { CheckBox } from '../..';
-import { StackedItemsGrid } from '../StackedItemsGrid';
-import { StackedRowItemValue } from '../StackedItemsGrid/types';
+import { colors } from '@shared/lib';
+
+import { CheckBox, YStack } from '../..';
+import {
+  StackedItemsGrid,
+  type StackedRowItemValue,
+  type Item,
+} from '../StackedItemsGrid';
+
+type StackedCheckboxConfig = {
+  rows: Array<Item>;
+  options: Array<Item>;
+};
+
+type Values = {
+  rowId: string;
+  optionIds: string[];
+}[];
 
 type Props = {
-  value: Array<string[]>;
-  onChange: (value: Array<string[]>) => void;
+  values: Values | null;
+  onChange: (value: Values | null) => void;
   config: StackedCheckboxConfig;
+  textReplacer: (markdown: string) => string;
 };
 
-const StackedCheckboxItem: FC<Props> = ({ value = [], onChange, config }) => {
-  const onValueChange = (option: StackedRowItemValue, itemIndex: number) => {
-    if (!value[itemIndex]?.length) {
-      value[itemIndex] = [];
-    }
+const StackedCheckboxItem: FC<Props> = ({
+  values,
+  onChange,
+  config,
+  textReplacer,
+}) => {
+  const { options, rows } = config;
 
-    if (value[itemIndex].includes(option.name)) {
-      value[itemIndex] = value[itemIndex].filter(
-        valueItem => valueItem !== option.name,
-      );
+  const memoizedOptions = useMemo(() => {
+    return options.map(option => {
+      return {
+        ...option,
+        tooltip: textReplacer(option.tooltip || ''),
+      };
+    });
+  }, [options, textReplacer]);
+
+  const memoizedRows = useMemo(() => {
+    return rows.map(row => {
+      return {
+        ...row,
+        tooltip: textReplacer(row.tooltip || ''),
+      };
+    });
+  }, [rows, textReplacer]);
+
+  const onValueChange = (option: StackedRowItemValue, row: Item) => {
+    const rowId = row.id;
+    const optionId = option.id;
+    const rowInitialValue = {
+      rowId,
+      optionIds: [optionId],
+    };
+
+    if (!values) {
+      const value = [rowInitialValue];
+
+      onChange(value);
     } else {
-      value[itemIndex].push(option.name);
-    }
+      const clonedValues = [...values];
+      let foundRow = false;
+
+      for (let key in clonedValues) {
+        const currentRow = clonedValues[key];
+        if (currentRow.rowId === rowId) {
+          foundRow = true;
+          if (currentRow.optionIds.includes(optionId)) {
+            currentRow.optionIds = currentRow.optionIds.filter(
+              id => id !== optionId,
+            );
+          } else {
+            currentRow.optionIds.push(optionId);
+          }
+        }
+      }
+      if (!foundRow) {
+        clonedValues.push(rowInitialValue);
+      }
```

Comment:
Instead of this for-loop and if-statement, you can do something like these:
```
const row = value[rowId] ?? {};
const optionIds = row?.optionIds ?? [];

const newOptionIds = [...optionIds];

if (optionIds.includes(optionId)) {
  newOptionIds = optionIds.filter(
    id => id !== optionId,
  );
} else {
  newOptionIds = [...newOptionIds, optionId];
}

const newValue = {
  ...values,
  [rowId]: newOptionIds,
};

```